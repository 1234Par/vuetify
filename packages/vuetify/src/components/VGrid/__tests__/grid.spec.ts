// Component factories
import grid from '../grid'

// Utilities
import {
  mount,
  MountOptions,
  Wrapper
} from '@vue/test-utils'
import { functionalContext } from '../../../../test'
import Vue, { VueConstructor } from 'vue'

describe('grid.ts', () => {
  type Instance = InstanceType<VueConstructor<{
    id: string
    tag: string
  } & Vue>>
  let mountFunction: (name?: string, options?: MountOptions<Instance>) => Wrapper<Instance>

  beforeEach(() => {
    mountFunction = (name = 'grid', options = {} as MountOptions<Instance>) => {
      return mount(grid(name), options)
    }
  })

  it('should render', () => {
    const wrapper = mountFunction()

    expect(wrapper.classes('grid')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with any class name', () => {
    const wrapper = mountFunction('flex')

    expect(wrapper.classes('grid')).toBeFalsy()
    expect(wrapper.classes('flex')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with any id', () => {
    const wrapper = mountFunction('grid', {
      context: {
        props: {
          id: 'test'
        }
      }
    })

    expect(wrapper.classes('grid')).toBeTruthy()
    expect(wrapper.attributes('id')).toBe('test')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with any tag', () => {
    const wrapper = mountFunction('grid', {
      context: {
        props: {
          tag: 'span'
        }
      }
    })

    expect(wrapper.element.tagName).toBe('SPAN')
    expect(wrapper.classes('grid')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
