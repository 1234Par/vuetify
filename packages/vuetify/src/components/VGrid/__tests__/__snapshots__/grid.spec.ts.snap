// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid.ts should render 1`] = `

<div class="grid">
</div>

`;

exports[`grid.ts should render with any class name 1`] = `

<div class="flex">
</div>

`;

exports[`grid.ts should render with any id 1`] = `

<div class="grid"
     id="test"
>
</div>

`;

exports[`grid.ts should render with any tag 1`] = `

<span class="grid">
</span>

`;
